src/Form/Field.php

```diff
@@ -255,6 +255,30 @@ public function isRequired(): bool
         return $this->required ?? false;
     }
 
+    protected function isSaveable(): bool
+    {
+        if ($this->isRequired() === true && $this->save() === true && $this->isEmpty() === true) {
+            // check the data of the relevant fields if there is a when option
+            if (empty($this->when) === false && is_array($this->when) === true) {
+                $input = $this->attrs['input'] ?? [];
+
+                if (empty($input) === false) {
+                    foreach ($this->when as $field => $value) {
+                        // if the input data doesn't have data for when fields
+                        // or doesn't match, that means is not required and return true
+                        if (($input[$field] ?? '') !== $value) {
```

Comment:
This is a clever hack, but I'm not sure if it's a good idea to rely on that logic to be true. Whether a value is in the `$input` is just an implementation detail of the `Form` class in the end.

I'd say the most solid solution would be to pass the `$fields` collection to the `Field` constructor so that each field has access to all other fields directly. It can then ask the other fields whether their value matches.

Reply:
Good idea! Probably way more robust, but for sure easier to understand later on.

Reply:
I liked the fiction if I understood correctly, but the point that stuck in my head is this:

We will send all the fields into the constructor of the field, but the first field will not know about the next field. Because the second field has not yet been added/appended in the `$fields` parameter sent to the first field.

Perhaps there may be issues with strict comparisons, but it was more accurate to do business on `input` values.

I'm thinking wrong?

Reply:
The `$fields` variable is an object, which is always passed by reference by PHP. So all properties like the array of fields inside the `$fields` object are only stored in one place and not copied. This means that all fields will have access to the full collection of fields as long as the *access* inside the field happens after all fields have been initialized (= accessing other fields inside the `Field` constructor will indeed not work, but in other methods that are called later it will).

Reply:
I know, but our `validate()`  method that  works within the constructor. I tried but couldn't be successful. I'm still not sure 🤔

https://github.com/getkirby/kirby/blob/3.4.0/src/Form/Field.php#L63

Can I leave this implement to you after last commit? @lukasbestle

Reply:
We could likely move this `validate` call to the `errors` method, which would lazily call it if the errors are not already defined.